website/docs/plugin/framework/functions/errors.mdx

```diff
@@ -0,0 +1,149 @@
+quickly. You should only concatenate a `FunctError` and never replace or
+remove information it.
+
+The next section will detail the concepts and typical behaviors of
+function error, while the final section will outline the typical methods for
+working with function error, using functionality from the available
+[`function` package](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function).
+
+## Function Error Concepts
+
+### Text
+
+`Text` is a practitioner-oriented description of the problem. This should
+contain sufficient detail to provide both general and more specific information
+regarding the issue. For example "Error executing function: foo can only contain
+letters, numbers, and digits."
+
+### FunctionArgument
+
+`FunctionArgument` identifies the specific function argument position that caused the
+error. Only errors that pertain to a function argument will include this information.
+
+### Working With Existing Function Errors
+
+#### ConcatFuncErrors
+
+When receiving `function.FuncError` from a function or method, such as
+`Run()`, these should typically be concatenated with the
+response function error for the method. This can be accomplished with the
+[`ConcatFuncErrors(in ...*FuncError)` method](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#ConcatFuncErrors).
+
+For example:
+
+```go
+func (f *ExampleFunction) Run(ctx context.Context, req function.RunRequest, resp *function.RunResponse) {
+    var boolArg bool
+    var stringArg string
+
+    resp.Error = function.ConcatFuncErrors(resp.Error, req.Arguments.Get(ctx, &boolArg, &stringArg))
+
+    // ... other logic ...
+}
+```
+
+This method automatically ignores `nil` function errors.
+
+### Creating Function Errors
+
+To craft the message of a function error, it is recommended to use sufficient
+detail to convey both the cause of the error and as much contextual,
+troubleshooting, and next action information as possible. These details can
+use newlines for easier readability where necessary.
+
+#### NewFuncError
+
+When creating function errors that affect an entire data source, provider, or
+resource, and where a `function.FunctionError` is already available such as within
+a response type, the [`ConcatFuncErrors(in ...*FuncError)` function](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#NewFuncError.AddError) can
+be used with the [`NewFuncError(text string)` function](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#NewFuncError) to concatenate
+a new function error.
+
+For example:
+
+```go
+func (f *ExampleFunction) Run(ctx context.Context, req function.RunRequest, resp *function.RunResponse) {
+    // ... prior logic ...
+
+	val, err := // operation that may return an error
+
+	if err != nil {
+        resp.Error = ConcatFuncErrors(resp.Error, function.NewFuncError("Error performing operation: " + err.Error()))
+        return
+    }
+
+    // ... further logic ...
+}
```

Comment:
nit: Go formatting 🙂 (does our website auto-fix the formatting? I wouldn't be surprised if it did)

Same formatting is a little off for some of the other examples (at least in GitHub)

Reply:
Ah, the old tabs/spaces fun and games. Fixed throughout the docs.